h2o-algos/src/main/java/hex/util/LinearAlgebraUtils.java

```diff
@@ -361,6 +361,9 @@ public static int numColsExp(Frame fr, boolean useAllFactorLevels) {
       for (int j = 0; j < ny; j++) {
         double yyij = i==j ? diagYY[i] : 0;
         uu[i][j] = (yyij - diagYY[i] * diagYY[j] / nTot) / (nVars * Math.sqrt(diagYY[i] * diagYY[j]));
+        if (Double.isNaN(uu[i][j])) {
+          uu[i][j] = 0;
```

Comment:
Not sure about this one, might be better to throw an exception instead of masking an error.